gmlc/networking/TcpOperations.h

```diff
@@ -22,16 +22,28 @@ class io_context;
 }  // namespace asio
 
 namespace gmlc::networking {
-/** establish a connection to a server by as associated timeout*/
-TcpConnection::pointer makeConnection(
+
+/** establish a connection to a server by as associated optional timeout
+@param[in] io_context the context to establish the connection
+@param[in] connection the address of the connection to establish
```

Comment:
I feel like this argument should be called `address`, though that is used in the other overload already; maybe `host`? That would fit if the address given to the other function is roughly the equivalent of a url (`scheme://host:port/path?query`).